lib/util/trip/drive_card.dart

```diff
@@ -1,75 +1,121 @@
 import 'package:flutter/material.dart';
-import 'package:motis_mitfahr_app/util/custom_timeline_theme.dart';
-
 import 'package:motis_mitfahr_app/util/trip/trip_card.dart';
+import 'package:motis_mitfahr_app/rides/models/ride.dart';
+import 'package:motis_mitfahr_app/util/trip/trip_card_state.dart';
 import '../../drives/models/drive.dart';
 import '../../drives/pages/drive_detail_page.dart';
-import 'package:timelines/timelines.dart';
-
-import '../locale_manager.dart';
-import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import '../supabase.dart';
+import 'package:motis_mitfahr_app/util/trip/trip_overview.dart';
+import 'package:motis_mitfahr_app/util/own_theme_fields.dart';
 
 class DriveCard extends TripCard<Drive> {
   const DriveCard(super.trip, {super.key});
 
+  @override
+  State<DriveCard> createState() => _DriveCardState();
+}
+
+class _DriveCardState extends TripCardState<DriveCard> {
+  Drive? _drive;
+  bool _fullyLoaded = false;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _drive = widget.trip;
+      super.trip = _drive;
+    });
+    loadDrive();
+  }
+
+  @override
+  void didUpdateWidget(DriveCard oldWidget) {
+    if (!trip!.equals(widget.trip)) {
+      loadDrive();
+    }
+    super.didUpdateWidget(oldWidget);
+  }
+
+  Future<void> loadDrive() async {
+    Map<String, dynamic> data = await SupabaseManager.supabaseClient.from('drives').select('''
+      *,
+      rides(
+        *,
+        rider: rider_id(*)
+      )
+    ''').eq('id', widget.trip.id).single();
+    if (mounted) {
+      setState(() {
+        _drive = Drive.fromJson(data);
+        super.trip = _drive;
+        _fullyLoaded = true;
+      });
+    }
+  }
+
+  @override
+  Widget buildRightSide() {
+    return TripOverview(super.trip!).buildSeatIndicator(context, super.trip!);
+  }
+
+  Color pickBannerColor() {
+    if (!_fullyLoaded) {
+      return Theme.of(context).cardColor;
+    } else {
+      if (_drive!.cancelled) {
+        return Theme.of(context).colorScheme.error;
+      } else if (_drive!.rides!.any((ride) => ride.status == RideStatus.pending)) {
```

Comment:
why would it be disabled in this case? It could maybe be yellow or blue or something (primary for example)